pennylane/ops/op_math/pow.py

```diff
@@ -216,8 +216,8 @@ def __init__(self, base=None, z=1, id=None):
             if (base_pauli_rep := getattr(self.base, "_pauli_rep", None)) and (
                 self.batch_size is None
             ):
-                pr = qml.pauli.PauliSentence({})
-                for _ in range(self.z):
+                pr = base_pauli_rep
+                for _ in range(self.z - 1):
                     pr = pr * base_pauli_rep
```

Comment:
Just a side comment: Looking at this, we may want to implement `PauliSentence.__pow__` at some point, using the structure of Pauli words, no? :)

Reply:
Indeed! I think that was on the plan, but we backlogged it due to time restraints. Other than the self inverse property and the commutation relations, do you know of any other property that would help speed up the implementation of the pow method? 

If not, the feature might just be taking those lines of code above and transferring them to a `__pow__` method in PS and PW.

Reply:
moved to slack to continue ^^